storage/tokudb/hatoku_hton.cc

```diff
@@ -919,16 +919,18 @@ static int tokudb_rollback(handlerton * hton, THD * thd, bool all) {
     TOKUDB_DBUG_RETURN(0);
 }
 
-static bool tokudb_sync_on_prepare(void) {
+static bool tokudb_sync_on_prepare(THD* thd) {
     TOKUDB_TRACE_FOR_FLAGS(TOKUDB_DEBUG_XA, "enter");
-    // skip sync of log if fsync log period > 0
-    if (tokudb::sysvars::fsync_log_period > 0) {
-        TOKUDB_TRACE_FOR_FLAGS(TOKUDB_DEBUG_XA, "exit");
-        return false;
-    } else {
-        TOKUDB_TRACE_FOR_FLAGS(TOKUDB_DEBUG_XA, "exit");
-        return true;
+    bool r = true;
+    // skip sync of log if fsync log period > 0 or if
+    // client durability during 2PC has been set to ignore, usually because
+    // binlog coordinator is in use and performing group commit
+    if (tokudb::sysvars::fsync_log_period > 0 ||
```

Comment:
Purely stylistically, I prefer to set bool variables directly instead of if (condition) foo = false; else foo = true; i.e, r = tokudb::sysvars::fsync_log_period <= 0 && !thd_get ..

Up to you to decide whether you want to make this change